pkg/provider/gitlab/gitlab.go

```diff
@@ -137,22 +145,25 @@ func (g *Gitlab) NewClient(ctx context.Context, store esv1beta1.GenericStore, ku
 
 	var err error
 
-	// Create client options
+	// Create projectClient options
 	var opts []gitlab.ClientOptionFunc
 	if cliStore.store.URL != "" {
 		opts = append(opts, gitlab.WithBaseURL(cliStore.store.URL))
 	}
+
 	// ClientOptionFunc from the gitlab package can be mapped with the CRD
 	// in a similar way to extend functionality of the provider
 
-	// Create a new Gitlab client using credentials and options
+	// Create a new Gitlab projectClient using credentials and options
 	gitlabClient, err := gitlab.NewClient(string(cliStore.credentials), opts...)
 	if err != nil {
 		return nil, err
 	}
 
-	g.client = gitlabClient.ProjectVariables
+	g.projectClient = gitlabClient.ProjectVariables
+	g.groupClient = gitlabClient.GroupVariables
 	g.projectID = cliStore.store.ProjectID
+	g.groupIDs = cliStore.store.GroupIDs
```

Comment:
several group-ids are supported ?

Reply:
a) Intention - gitlab supports multiple group layers each with their own variables

Reply:
thx for the info @domizei385

Reply:
> Due to the amount of calls to recursively determine parent groups of a project, the groups to pull in secrets from will have to be manually configured (through groupIDs).

IIRC Group variables aren't inherited from an API perspective so we do have to call `GET /groups/:id/variables` for every group.
I don't see a problem with walking up the tree using the `parent_id` of a given group and fetch the vars for each group. From a user perspective i'd say this is desirable over specifying the all the group IDs: A user just wants to specify the group id he/she cares about. In fact group layouts in GitLab change over time and from a user's perspective i wouldn't want to maintain/synchronize state changes for both GitLab groups and external-secrets secrets.

Reply:
With the effort I meant that I would have to pull the data from the "project info" endpoint to get the first group id, then traversing up the tree through further "group info" calls until reaching the root. Eventually I have the full list, but having to do this many calls for each secret seems excessive.
I guess I could live with validating whether the given ids represent a valid inheritance tree.

Explicitly configuring the group ids forces the user to think about permissions for each group in question.
NOTE: While Gitlab exposes group variables (of groups you have limited access to) to a project deployment it is not returning those through the API.

Reply:
I added an option now to automatically discover project groups.

Reply:
> traversing up the tree through further "group info" calls

I'm afraid this is needed anyway: The [list-project-groups](https://docs.gitlab.com/ee/api/projects.html#list-a-projects-groups) endpoint doesn't seem to return the `parent_id`. without that information we don't know how to handle value precedence (leaf > root)

Reply:
As mentioned in the other comment: I kind of read somewhere that the response is already ordered, but am unable to find it now, so must assume the worst.